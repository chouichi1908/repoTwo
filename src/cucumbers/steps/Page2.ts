import { Given, When, Then} from '@cucumber/cucumber';
import { Selector, RequestMock} from 'testcafe';

// require is ok
// import is failed 
// maybe the testcafe use common.js 
const mockData = require('../../mock/weather.json')


const mockUrl = new RegExp(/^https:\/\/api.openweathermap.org\/data\/2.5\/weather/)
const mock = RequestMock().onRequestTo(mockUrl).respond(mockData,200,{
  'Access-Control-Allow-Origin': '*'
})


Given('Open home page',async (t)=>{
    await t.navigateTo('http://localhost:3000/').addRequestHooks(mock)
} )

Then('current page is Home',async (t)=>{
  const location = await t.eval(()=>window.location)
  await t.expect(location.pathname).eql('/')
})

When('click page2 link',async (t)=>{
  const link = Selector('a').withText('Page2')
  await t.click(link)
})

Then('current page is Page2',async (t)=>{
  const location = await t.eval(()=>window.location)
  await t.expect(location.pathname).eql('/currentweather')
})

When('input text tokyo',async (t)=>{
  const input = Selector('input')
  const confirm = Selector('button') 
  await t.typeText(input,'東京').click(confirm)
})

Then('confirm mock data',async (t)=>{
  const cityName = Selector('.item').nth(0);
  // const error = Selector('p');
  // await t.expect(error.textContent).contains('エラーが発生しました。')
  await t.expect(cityName.textContent).contains('Osamu')
})

When('click home link',async (t)=>{

  const link = Selector('a').withText('Home')
  await t.click(link)

})


