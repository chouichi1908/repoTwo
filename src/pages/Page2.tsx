import Header from '../components/Header'
import Footer from '../components/Footer';
import Search from '../components/Search';
import React from 'react';


function Page2(){


    return(
        <React.Fragment>
            <Header />
            <h1 role='contentinfo'>This is page2</h1>
            <Search type='weather' />
            <img src="/logo512.png" alt="log"/>
            <Footer />
        </React.Fragment>
        
    )

}


export default Page2;
