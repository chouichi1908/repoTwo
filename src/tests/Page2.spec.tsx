import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Page2 from "../pages/Page2";
import { BrowserRouter as Router } from "react-router-dom";
import { rest } from "msw";
import { setupServer } from "msw/node";
import { Provider } from "react-redux";
import store from "../stores";
import mockData from "../mock/weather.json";

export const handlers = [
  rest.get(
    "https://api.openweathermap.org/data/2.5/weather",
    (req, res, ctx) => {
      const city = req.url.searchParams.get("q");
      if (city == "東京") {
        return res(ctx.json(mockData), ctx.delay(150));
      } else if (city == "上海") {
        return res(ctx.status(404, "city name wrong."));
      }
    }
  ),
];

const server = setupServer(...handlers);

// Enable API mocking before tests.
beforeAll(() => server.listen());

// Reset any runtime request handlers we may add during the tests.
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());

jest.spyOn(console, "error");

describe("Page2 tests", () => {
  test("Snapshot test", () => {
    const dom = render(
      <Provider store={store}>
        <Router>
          <Page2 />
        </Router>
      </Provider>
    );
    expect(dom).toMatchSnapshot();
  });

  test("user type city and click comfirm", async () => {
    render(
      <Provider store={store}>
        <Router>
          <Page2 />
        </Router>
      </Provider>
    );
    const input = screen.getByRole("textbox");
    userEvent.type(input, "東京");
    expect(screen.getByRole("textbox")).toHaveValue("東京");
    userEvent.click(screen.getByRole("button", { name: /Submit/i }));
    expect(await screen.findByText(/都市名:Osamu/i)).toBeInTheDocument();
  });

  test("test error when the city name is wrong", async () => {
    render(
      <Provider store={store}>
        <Router>
          <Page2 />
        </Router>
      </Provider>
    );
    const input = screen.getByRole("textbox");
    userEvent.type(input, "上海");
    expect(screen.getByRole("textbox")).toHaveValue("上海");
    userEvent.click(screen.getByRole("button", { name: /Submit/i }));
    expect(
      await screen.findByText(/エラーが発生しました。/i)
    ).toBeInTheDocument();
    expect(console.error).toHaveBeenCalledWith("エラーが発生しました。");
  });
});
