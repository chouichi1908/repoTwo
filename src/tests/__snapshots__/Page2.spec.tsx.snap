// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page2 tests Snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui inverted vertical segment"
        style="padding: 1.2em 0em;"
      >
        <div
          class="ui container"
        >
          <div
            class="ui inverted stackable grid"
          >
            <div
              class="row"
            >
              <div
                class="three wide column"
              >
                <a
                  class="ui blue inverted header"
                  href="/"
                >
                  Home
                </a>
              </div>
              <div
                class="three wide column"
              >
                <a
                  class="ui olive inverted header"
                  href="/currentweather"
                >
                  Page2
                </a>
              </div>
              <div
                class="three wide column"
              >
                <a
                  class="ui violet inverted header"
                  href="/forecasteachthreehours"
                >
                  Page3
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="min-height: 78vh;"
      >
        <h1
          role="contentinfo"
        >
          This is 
          page2
        </h1>
        <form
          action="#"
        >
          <input
            name="city"
            value=""
          />
          <button
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <div
        class="ui inverted vertical segment"
        style="padding: 1.2em 0em;"
      >
        <div
          class="ui container"
        >
          <div
            class="ui inverted stackable grid"
          >
            <div
              class="row"
            >
              <div
                class="three wide column"
              >
                <h4
                  class="ui violet inverted header"
                >
                  关于我们
                </h4>
                <div
                  class="ui inverted link list"
                  role="list"
                >
                  <a
                    class="item"
                    role="listitem"
                  >
                    网站地图
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    联系我们
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    合作公司
                  </a>
                </div>
              </div>
              <div
                class="three wide column"
              >
                <h4
                  class="ui violet inverted header"
                >
                  友情链接
                </h4>
                <div
                  class="ui inverted link list"
                  role="list"
                >
                  <a
                    class="item"
                    role="listitem"
                  >
                    Google
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    Youtube
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    Instagram
                  </a>
                </div>
              </div>
              <div
                class="right floated six wide column"
              >
                <h4
                  class="ui violet inverted header"
                >
                  版权声明
                </h4>
                <p>
                  Copyright © Dynamic Gravity 
                  2021
                  .
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui inverted vertical segment"
      style="padding: 1.2em 0em;"
    >
      <div
        class="ui container"
      >
        <div
          class="ui inverted stackable grid"
        >
          <div
            class="row"
          >
            <div
              class="three wide column"
            >
              <a
                class="ui blue inverted header"
                href="/"
              >
                Home
              </a>
            </div>
            <div
              class="three wide column"
            >
              <a
                class="ui olive inverted header"
                href="/currentweather"
              >
                Page2
              </a>
            </div>
            <div
              class="three wide column"
            >
              <a
                class="ui violet inverted header"
                href="/forecasteachthreehours"
              >
                Page3
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="min-height: 78vh;"
    >
      <h1
        role="contentinfo"
      >
        This is 
        page2
      </h1>
      <form
        action="#"
      >
        <input
          name="city"
          value=""
        />
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <div
      class="ui inverted vertical segment"
      style="padding: 1.2em 0em;"
    >
      <div
        class="ui container"
      >
        <div
          class="ui inverted stackable grid"
        >
          <div
            class="row"
          >
            <div
              class="three wide column"
            >
              <h4
                class="ui violet inverted header"
              >
                关于我们
              </h4>
              <div
                class="ui inverted link list"
                role="list"
              >
                <a
                  class="item"
                  role="listitem"
                >
                  网站地图
                </a>
                <a
                  class="item"
                  role="listitem"
                >
                  联系我们
                </a>
                <a
                  class="item"
                  role="listitem"
                >
                  合作公司
                </a>
              </div>
            </div>
            <div
              class="three wide column"
            >
              <h4
                class="ui violet inverted header"
              >
                友情链接
              </h4>
              <div
                class="ui inverted link list"
                role="list"
              >
                <a
                  class="item"
                  role="listitem"
                >
                  Google
                </a>
                <a
                  class="item"
                  role="listitem"
                >
                  Youtube
                </a>
                <a
                  class="item"
                  role="listitem"
                >
                  Instagram
                </a>
              </div>
            </div>
            <div
              class="right floated six wide column"
            >
              <h4
                class="ui violet inverted header"
              >
                版权声明
              </h4>
              <p>
                Copyright © Dynamic Gravity 
                2021
                .
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
